R/revision_analysis.R

```diff
@@ -0,0 +1,209 @@
+  }
+  revision_behavior <- revision_behavior %>%
+    mutate(across(c(version, time_value), list(int = as.integer))) %>%
+    mutate(lag = version_int - time_value_int) %>% # nolint: object_usage_linter
+    group_by(across(all_of(keys))) %>% # group by all the keys
+    select(-time_value_int, -version_int) %>%
+    summarize(
+      n_revisions = dplyr::n() - 1,
+      min_lag = min(lag), # nolint: object_usage_linter
+      max_lag = max(lag), # nolint: object_usage_linter
+      max_change = suppressWarnings(max(!!arg[[1]], na.rm = TRUE) - min(!!arg[[1]], na.rm = TRUE)),
+      max_rel_change = (max_change) / suppressWarnings(max(!!arg[[1]], na.rm = TRUE)), # nolint: object_usage_linter
+      time_to = time_to_x_percent(lag, !!arg[[1]], percent = percent_final_value)
+    ) %>%
+    ungroup() %>%
+    mutate(
+      # TODO the units here may be a problem
+      min_lag = as.difftime(min_lag, units = "days"), # nolint: object_usage_linter
+      max_lag = as.difftime(max_lag, units = "days"), # nolint: object_usage_linter
+      time_to_pct_final = as.difftime(time_to, units = "days") # nolint: object_usage_linter
+    ) %>%
+    select(-time_to)
+  if (print_inform) {
+    total_num <- nrow(revision_behavior) # nolint: object_usage_linter
+    cli_inform("Number of revisions:")
+    cli_inform("Min lag (time to first version):")
+    difftime_summary(revision_behavior$min_lag)
+    if (!drop_nas) {
+      total_na <- nrow(epi_arch$DT %>% filter(is.na(!!arg[[1]]))) # nolint: object_usage_linter
+      cli_inform("Fraction of all versions that are `NA`:")
+      cli_li(num_percent(total_na, nrow(epi_arch$DT)))
+    }
+    total_num_unrevised <- sum(revision_behavior$n_revisions == 0) # nolint: object_usage_linter
+    cli_inform("No revisions:")
+    cli_li(num_percent(total_num_unrevised, total_num))
+    total_quickly_revised <- sum( # nolint: object_usage_linter
+      revision_behavior$max_lag <=
+        as.difftime(quick_revision, units = "days")
+    )
+    cli_inform("Quick revisions (last revision within {quick_revision}
+{units(quick_revision)} of the `time_value`):")
+    cli_li(num_percent(total_quickly_revised, total_num))
+    total_barely_revised <- sum( # nolint: object_usage_linter
+      revision_behavior$n_revisions <=
+        few_revisions
+    )
+    cli_inform("Few revisions (At most {few_revisions} revisions for that `time_value`):")
+    cli_li(num_percent(total_barely_revised, total_num))
+    cli_inform("")
+    cli_inform("Changes in Value:")
+
+    real_revisions <- revision_behavior %>% filter(n_revisions > 0) # nolint: object_usage_linter
+    n_real_revised <- nrow(real_revisions) # nolint: object_usage_linter
+    rel_change <- sum( # nolint: object_usage_linter
+      real_revisions$max_rel_change <
+        rel_change_threshold,
+      na.rm = TRUE
+    ) + sum(is.na(real_revisions$max_rel_change))
+    cli_inform("Less than {rel_change_threshold} change in relative value (only from the revised subset):")
```

Comment:
question: earlier, we found some seemingly-contradictory results between this and the time-to-near-latest results for some example.  But I'm trying on `archive_cases_dv_subset` and not seeing an issue; did you figure this out?

Reply:
I did yes. Fixed it when I was adding tests, there was an inequality pointing the wrong direction